providers/catalog_test.go

```diff
@@ -93,6 +93,31 @@ var testCases = []struct { // nolint
 		},
 		expectedErr: nil,
 	},
+	{
+		provider:    Zendesk,
+		description: "Zendesk provider config with valid & invalid substitutions",
+		substitutions: map[string]string{
+			"nonexistentvar": "test",
+		},
+		expected: &ProviderInfo{
+			Support: Support{
+				Read:      false,
+				Write:     false,
+				BulkWrite: false,
+				Subscribe: false,
+				Proxy:     true,
+			},
+			AuthType: Oauth2,
+			OauthOpts: OauthOpts{
+				AuthURL:                   "https://d3v-ampersand.zendesk.com/oauth/authorizations/new",
+				TokenURL:                  "https://d3v-ampersand.zendesk.com/oauth/token",
+				ExplicitWorkspaceRequired: true,
+				ExplicitScopesRequired:    false,
```

Comment:
Zendesk does support the following scopes, but it was currently set to false to minimize potential causes for error.

- tickets
- users
- auditlogs (read only)
- organizations
- hc
- apps
- triggers
- automations
- targets
- webhooks
- macros
- requests
- satisfaction_ratings
- dynamic_content
- any_channel (write only)
- web_widget (write only)